src/k8s-engine/lima.ts

```diff
@@ -679,28 +747,129 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
+      console.log(`VDE tools install required: ${ command }`);
+      commands.push(command);
+    } finally {
+      commands.push(`rm -fr ${ workdir }`);
+    }
+  }
 
-        console.log(`VDE tools install required: ${ command }`);
-        sudo.exec(command, { name: 'Rancher Desktop' }, (error, stdout, stderr) => {
-          if (stdout) {
-            console.log(`Prompt for sudo: stdout: ${ stdout }`);
-          }
-          if (stderr) {
-            console.log(`Prompt for sudo: stderr: ${ stderr }`);
-          }
-          if (error) {
-            reject(error);
-          } else {
-            resolve();
-          }
+  protected async createLimaSudoersFile(commands: Array<string>) {
+    try {
+      await this.lima('sudoers', '--check');
+      console.log(`lima sudoers --check is ok`);
+    } catch (_) {
+      // Here we have to run `lima sudoers` as non-root and grab the output, and then
+      // copy it to the target sudoers file as root
+      const { stdout : data } = await childProcess.spawnFile('limactl', ['sudoers'],
+        {
+          stdio: ['inherit', 'pipe', console],
+          env:   this.limaEnv
         });
+      const tmpFile = path.join(os.tmpdir(), 'sudoers.txt');
+
+      await fs.promises.writeFile(tmpFile, data.toString());
+      console.log(`need to limactl sudoers, get data from ${ tmpFile }`);
+      commands.push(`cp ${ tmpFile } ${ LIMA_SUDOERS_LOCATION } && rm -f ${ tmpFile }`);
+    }
+  }
+
+  protected async ensureRunLimaLocation(commands: Array<string>) {
+    let dirInfo;
+    let dirExists;
+
+    try {
+      dirInfo = await fs.promises.stat(RUN_LIMA_LOCATION);
+
+      if (dirInfo && dirInfo.uid === 0 && (dirInfo.mode & 2) === 0) {
+        return;
+      }
+      dirExists = true;
+    } catch (err) {
+      dirInfo = null;
+      if (err.code === 'ENOENT') {
+        dirExists = false;
+      } else {
+        console.log(`Unexpected situation with ${ RUN_LIMA_LOCATION }, stat => error ${ err }`, err);
+        throw err;
+      }
+    }
+    if (!dirInfo || !dirExists) {
+      commands.push(`mkdir -p ${ RUN_LIMA_LOCATION }`);
+      commands.push(`chmod 755 ${ RUN_LIMA_LOCATION }`);
+    } else if (dirInfo.uid !== 0) {
+      // Safe assumption that root is always user ID 0
+      commands.push(`chown -R root:wheel ${ RUN_LIMA_LOCATION }`);
+      commands.push(`chmod -R u-w ${ RUN_LIMA_LOCATION }`);
+    } else {
+      commands.push(`chmod -R u-w ${ RUN_LIMA_LOCATION }`);
+    }
+  }
+
+  /**
+   * Use the sudo-prompt library to run the script as root
+   * @param command: Name of an executable shell script (could be a binary, but no args are provided)
+   * @protected
+   */
+  protected async sudoExec(command: string) {
+    try {
+      // Try directly running the command as sudo before prompting
+      const cp = await childProcess.spawnFile('sudo', ['--non-interactive', '--', command],
+        { stdio: ['ignore', 'pipe', 'pipe'] });
+
+      if (!cp.stderr) {
```

Comment:
That's a suspicious check: what if sudo succeeds, but the script we run emits to `stderr`? We'd end up running it twice?

If we require that the script we run be idempotent, we should document that.

Otherwise, we should use `sudo --non-interactive --validate` instead.

Reply:
I dropped this code. The `sudo-prompt` docs say it should respect cached credentials, and even if it doesn't, we're now only calling it once, with a script, rather than several times with individual commands.